concent_api/api-integration-multi-threads-test.py

```diff
@@ -0,0 +1,214 @@
+
+# !/usr/bin/env python3
+import os
+import sys
+import time
+from threading import Thread
+from typing import Optional
+
+import requests
+from golem_messages import message
+from golem_messages.factories.tasks import ReportComputedTaskFactory
+from golem_messages.message import Message
+from golem_messages.message.concents import ForceGetTaskResult
+from golem_messages.message.concents import ForceReportComputedTask
+from golem_messages.message.tasks import AckReportComputedTask
+from golem_messages.message.tasks import ReportComputedTask
+
+from api_testing_common import count_fails
+from api_testing_common import PROVIDER_PRIVATE_KEY
+from api_testing_common import REQUESTOR_PRIVATE_KEY
+from api_testing_common import api_request
+from api_testing_common import assert_condition
+from api_testing_common import create_signed_task_to_compute
+from api_testing_common import run_tests
+from api_testing_common import timestamp_to_isoformat
+from common.helpers import get_current_utc_timestamp
+from common.helpers import sign_message
+
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "concent_api.settings")
+
+REPORT_COMPUTED_TASK_SIZE = 10
+
+NUMBER_OF_TESTING_THREADS = 3  # changing this variable remember to correct number of expected responses
+MAXIMUM_WAITING_TIME_FOR_ALL_RESPONSES = 3  # seconds
+
+responses_global = []  # type: list
+
+
+def clear_responses():
+    global responses_global
+    responses_global.clear()
+
+
+def call_function_in_treads(
+    func,
+    number_of_threads: int,
+    cluster_url: str,
+    golem_message: Message,
+) -> None:
+    for i in range(number_of_threads):
+        t = Thread(target=func, args=(cluster_url, golem_message,))
+        t.start()
+
+
+def get_ack_report_computed_task(report_computed_task: ReportComputedTask) -> AckReportComputedTask:
+    ack_report_computed_task = AckReportComputedTask(report_computed_task=report_computed_task)
+    return ack_report_computed_task
```

Comment:
If you are creating messages this way please do this in one line:
```python
return AckReportComputedTask(report_computed_task=report_computed_task)
```

Reply:
done